csharp/src/Microsoft.ML.OnnxRuntime/Training/NativeTrainingMethods.cs

```diff
@@ -0,0 +1,256 @@
+                                                    IntPtr /*(OrtSessionRunOptions*)*/ runOptions  // can be null to use the default options
+                                                    );
+
+            public static DOrtOptimizerStep OrtOptimizerStep;
+
+            [UnmanagedFunctionPointer(CallingConvention.Winapi)]
+            public delegate IntPtr /*(ONNStatus*)*/ DOrtSetLearningRate(
+                                                    IntPtr /*(OrtTrainingSession*)*/ session,
+                                                    float learningRate
+                                                    );
+
+            public static DOrtSetLearningRate OrtSetLearningRate;
+
+            [UnmanagedFunctionPointer(CallingConvention.Winapi)]
+            public delegate IntPtr /*(ONNStatus*)*/ DOrtGetLearningRate(
+                                                    IntPtr /*(OrtTrainingSession*)*/ session,
+                                                    ref float learningRate
+                                                    );
+
+            public static DOrtGetLearningRate OrtGetLearningRate;
+
+            [UnmanagedFunctionPointer(CallingConvention.Winapi)]
+            public delegate IntPtr /*(ONNStatus*)*/ DOrtRegisterLinearLRScheduler(
+                                                    IntPtr /*(OrtTrainingSession*)*/ session,
+                                                    long warmupStepCount,
+                                                    long totalStepCount,
+                                                    float learningRate
+                                                    );
+            public static DOrtRegisterLinearLRScheduler OrtRegisterLinearLRScheduler;
+
+            [UnmanagedFunctionPointer(CallingConvention.Winapi)]
+            public delegate IntPtr /*(ONNStatus*)*/ DOrtSchedulerStep(
+                                                    IntPtr /*(OrtTrainingSession*)*/ session
+                                                    );
+            public static DOrtSchedulerStep OrtSchedulerStep;
+
+            [UnmanagedFunctionPointer(CallingConvention.Winapi)]
+            public delegate void DOrtReleaseTrainingSession(IntPtr /*(OrtSession*)*/session);
+            public static DOrtReleaseTrainingSession OrtReleaseTrainingSession;
+
+            [UnmanagedFunctionPointer(CallingConvention.Winapi)]
+            public delegate void DOrtReleaseCheckpointState(IntPtr /*(OrtSession*)*/session);
+            public static DOrtReleaseCheckpointState OrtReleaseCheckpointState;
+
+            #endregion TrainingSession API
+
+            public static byte[] GetPlatformSerializedString(string str)
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    return System.Text.Encoding.Unicode.GetBytes(str + Char.MinValue);
+                else
+                    return System.Text.Encoding.UTF8.GetBytes(str + Char.MinValue);
```

Comment:
>ystem.Text.Encoding.UTF8.GetBytes(str + Char.MinValue); [](http://example.com/codeflow?start=28&length=55)

This is a code duplicate. Also it is not a native entry point, so probably misplaced. Can we put this into helpers?